lib/resque.rb

```diff
@@ -187,6 +187,55 @@ def enqueue_front
     end
   end
 
+  attr_writer :per_worker_stats
+  def per_worker_stats
+    if defined? @per_worker_stats
+      @per_worker_stats
+    else
+      @per_worker_stats = true
+    end
+  end
+
+  attr_writer :track_starts
+  def track_starts
+    if defined? @track_starts
+      @track_starts
+    else
+      @track_starts = true
+    end
+  end
+
+  attr_writer :queues_in_names
+  def queues_in_names
+    if defined? @queues_in_names
+      @queues_in_names
+    else
+      @queues_in_names = true
+    end
+  end
+
+  attr_writer :blocking_reserve
+  def blocking_reserve
+    if defined? @blocking_reserve
+      @blocking_reserve
+    else
+      @blocking_reserve = false
+    end
+  end
```

Comment:
FEATURE: `blocking_reserve`

Set this flag to `true` to make `reserve` use `blpop`. The default value preserves the stock behavior.